deeplite/profiler/profiler.py

```diff
@@ -249,44 +253,30 @@ def display_status(self, other=None, print_mode='debug', short_print=True):
         """
         # NOTE: force pop layerwise summary and display in debug mode as it is quite long regardless
         # of the display filter function.
-        assert isinstance(print_mode, str) and hasattr(logger, print_mode.lower()), \
-            "'print_mode' needs to be a logger level"
-
-        status_dict = self.status_to_dict(to_value=False)
-        layerwise_summary_1 = status_dict.pop('layerwise_summary', None)
-        status_dict = self.display_status_filter_func(status_dict)
-        status_dict['backend'] = self.backend
-        status_dict['name'] = self.name
-
-        if other is not None:
-            other_status_dict = other.status_to_dict(to_value=False)
-            layerwise_summary_2 = other_status_dict.pop('layerwise_summary', None)
-            other_status_dict = self.display_status_filter_func(other_status_dict)
-            other_status_dict['backend'] = other.backend
-            other_status_dict['name'] = other.name
-            summary_str = make_two_models_summary_str(status_dict, other_status_dict, short_print=short_print)
-        else:
-            layerwise_summary_2 = None
-            summary_str = make_one_model_summary_str(status_dict, short_print=short_print)
-
-        if not short_print:
-            if layerwise_summary_1:
-                logger.debug(layerwise_summary_1.value)
-            if layerwise_summary_2:
-                logger.debug(layerwise_summary_2.value)
-        getattr(logger, print_mode.lower())(summary_str)
+        self.display.display_status(self, other, print_mode, short_print)
 
     def clone(self, model=None, data_splits=None, retain_status=False):
         # create a new instance instead of deepcopying stuff
         model = self.model if not model else model
         data_splits = self.data_splits if not data_splits else data_splits
-        new = type(self)(model, data_splits, display_status_filter_func=self.display_status_filter_func)
+        new = type(self)(model, data_splits, display=self.display)
         new._profiling_functions_register = deepcopy(self._profiling_functions_register)
         if not retain_status and new.model is not self.model:
             new.reset_status()
         new.backend = self.backend
         return new
 
+    def get_eval_pfr(self):
```

Comment:
A ProfilingFunctionRegister is a private entity, there should be no public functions returning one. Could you find a way of achieving the goal of this PR without having to expose the pfr (even if they are for eval)?